src/common/services/api/designations.service.js

```diff
@@ -185,6 +185,11 @@ class DesignationsService {
       .slice(0, 5)
       .join('/')
 
+    const underscorePosition = code.indexOf('_')
+    if (underscorePosition > 0) {
+      code = code.substring(0, underscorePosition)
+    }
```

Comment:
What you have is fine but it could be a one liner (which is maybe more computationally since it processes the whole string instead of just to the first `_`).
```suggestion
    [code] = code.split('_');
```

Reply:
Hmm, that one liner appears to break all the tests. It works if I do this...
```
const other = code.split('_')

    return campaignPage
      ? `/content/give/us/en/campaigns/${c}/${other[0]}/${campaignPage}.infinity.json`
      : `/content/give/us/en/designations/${c}/${other[0]}.infinity.json`
```

Reply:
Also, this works:
```
const [other] = code.split('_')

return campaignPage
      ? `/content/give/us/en/campaigns/${c}/${other}/${campaignPage}.infinity.json`
      : `/content/give/us/en/designations/${c}/${other}.infinity.json`
```

Reply:
Something in the test tooling must not support array destructuring (and other newer JS features). Oh well, that's fine.
```suggestion
    code = code.split('_')[0];
```
is the same thing.

I like that you gave the variable a new name since I like to try and keep things immutable. This codebase doesn't value that.

Reply:
Huh that's interesting. Maybe assignment is a different feature than variable declarations. Whatever. I'm fine with whatever. This is trivial.